apps/aecore/src/aec_tx_sign.erl

```diff
@@ -138,6 +139,19 @@ deserialize_from_binary(SignedTxBin) when is_binary(SignedTxBin) ->
     lager:debug("unpacked Tx: ~p", [Unpacked]),
     deserialize(Unpacked).
 
-serialize_for_client(#signed_tx{data = Tx, signatures = Sigs}) ->
+-spec serialize_for_client(json|message_pack, #header{}, aec_tx_sign:signed_tx()) ->
+                              binary() | map().
+serialize_for_client(Encoding, Header, #signed_tx{}=S) ->
+    {ok, Hash} = aec_headers:hash_header(Header),
+    serialize_for_client(Encoding, S, aec_headers:height(Header), Hash).
+
+serialize_for_client(message_pack, #signed_tx{}=S, BlockHeight, BlockHash) ->
+    GenericData = [#{<<"block_height">> => BlockHeight},
+                   #{<<"block_hash">> => BlockHash}],
+    aec_base58c:encode(transaction, msgpack:pack(serialize(S) ++ GenericData));
```

Comment:
The usage of msgpack here seems similar to `serialize_to_binary` though with `++ GenericData`. Are we sure user does not want same msgpack as in `serialize_to_binary` (as in Merkle tree of txs)?

Reply:
It had been a special requirement for the client APIs the transaction objects to include certain fields. Currently those are just the hash and height of the block that included the transaction but there will be more to come (transaction's hash for example).
Having this in mind, we can not have the same msgpack both for the gossip protocol and the client APIs